### ISSUE TITLE
[Bug]: TPCH Q5 sometime leads to the mo crash, especially on ARM

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#4036]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):0.5.0
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

When run TPCH Q5 on arm, sometimes the mo corrupt.

mysql> select
    ->         n_name,
    ->         sum(l_extendedprice * (1 - l_discount)) as revenue
    -> from
    ->         customer,
    ->         orders,
    ->         lineitem,
    ->         supplier,
    ->         nation,
    ->         region
    -> where
    ->         c_custkey = o_custkey
    ->         and l_orderkey = o_orderkey
    ->         and l_suppkey = s_suppkey
    ->         and c_nationkey = s_nationkey
    ->         and s_nationkey = n_nationkey
    ->         and n_regionkey = r_regionkey
    ->         and r_name = 'AMERICA'
    ->         and o_orderdate >= date '1994-01-01'
    ->         and o_orderdate < date '1994-01-01' + interval '1' year
    -> group by
    ->         n_name
    -> order by
    ->         revenue desc
    -> ;
ERROR 2013 (HY000): Lost connection to MySQL server during query

**Comment by JinHai-CN** [comment]
@aressu1985 Will running Q5 lead to MO crash on x86 server?

**Comment by aressu1985** [comment]
fixed:e23e00f1d01d43e33c509a6d239ccc38b5f93409
